src/controller/scan/base_controller.go

```diff
@@ -578,34 +583,59 @@ func (bc *basicController) GetStats(requester string) (*all.Stats, error) {
 }
 
 // makeRobotAccount creates a robot account based on the arguments for scanning.
-func (bc *basicController) makeRobotAccount(projectID int64, repository string, registration *scanner.Registration) (*model.Robot, error) {
+func (bc *basicController) makeRobotAccount(ctx context.Context, projectID int64, repository string, registration *scanner.Registration) (*robot.Robot, error) {
 	// Use uuid as name to avoid duplicated entries.
 	UUID, err := bc.uuid()
 	if err != nil {
 		return nil, errors.Wrap(err, "scan controller: make robot account")
 	}
 
-	resource := rbac.NewProjectNamespace(projectID).Resource(rbac.ResourceRepository)
-	robotReq := &model.RobotCreate{
-		Name:        fmt.Sprintf("%s-%s", registration.Name, UUID),
-		Description: "for scan",
-		ProjectID:   projectID,
-		Access: []*types.Policy{
-			{Resource: resource, Action: rbac.ActionPull},
-			{Resource: resource, Action: rbac.ActionScannerPull},
+	p, err := bc.pro.Get(ctx, projectID)
+	if err != nil {
+		return nil, errors.Wrap(err, "scan controller: make robot account")
+	}
+
+	robotReq := &robot.Robot{
+		Robot: model.Robot{
+			Name:        fmt.Sprintf("%s-%s", registration.Name, UUID),
+			Description: "for scan",
+			ProjectID:   projectID,
+			ExpiresAt:   -1,
```

Comment:
Can set the expiration time to the value in the system configuration here?

Reply:
will raise another PR to introduce duration per discussion with UI designer.